index.js

```diff
@@ -6,6 +6,7 @@ var {
 
 function JMON(data) {
   this.data = data;
+  this.currentData = JSON.parse(JSON.stringify(data));
```

Comment:
I was thinking of name this `this.initialData`
Any thoughts on this?

Reply:
Why are you stringifying and parsing this. Any specific reason for this?
I was thinking may be we can directly store it as data.

Since we know we accept only json objects.
I think for better we should have a handler here, which will check whether the input is a json or not, if its not it should throw an error.

What do you think about this?

Reply:
> Ok. yes `this.data` will be `this.initialData`
> Then what is this.currentData? new modified data?

I have updated the this.initialData  to inital Data and current Data to this.data